src/actions/__tests__/onPaste.ts

```diff
@@ -0,0 +1,85 @@
+import SimplePath from '../../@types/SimplePath'
+import { HOME_TOKEN } from '../../constants'
+import { getAllChildren } from '../../selectors/getChildren'
+import getThoughtById from '../../selectors/getThoughtById'
+import store from '../../stores/app'
+import { importDataActionCreator as importData } from '../importData'
+
+describe('onPaste', () => {
+  beforeEach(() => {
+    // Reset store to initial state before each test
+    store.dispatch({ type: 'CLEAR_THOUGHT_CACHE' })
+  })
+
+  it('should correctly import nested bulleted lists from Notion', () => {
+    const notionHtml = `
+      <meta charset='utf-8'><ul>
+        <li>a
+          <ul>
+            <li>m
+              <ul>
+                <li>x</li>
+              </ul>
+            </li>
+          </ul>
+        </li>
+        <li>b
+          <ul>
+            <li>m
+              <ul>
+                <li>y</li>
+              </ul>
+            </li>
+          </ul>
+        </li>
+      </ul>
+    `
+
+    const plainText = ''
+
+    // Dispatch the import action directly
+    store.dispatch(
+      importData({
+        path: [HOME_TOKEN] as SimplePath,
+        text: plainText,
+        html: notionHtml,
+        rawDestValue: '',
+        isEmText: false,
+      }),
+    )
+
+    // Get the root children
+    const rootChildren = getAllChildren(store.getState(), HOME_TOKEN)
+
+    // Verify the structure
+    expect(rootChildren.length).toBe(2)
+
+    // Find 'a' and 'b' thoughts
+    const aThought = rootChildren.find(id => getThoughtById(store.getState(), id)?.value === 'a')
+    const bThought = rootChildren.find(id => getThoughtById(store.getState(), id)?.value === 'b')
```

Comment:
To test the thought structure, use `exportContext` and assert the entire output. This is easier to read than walking the thought tree and asserting each level:

```ts
  const exported = exportContext(store.getState(), [HOME_TOKEN], 'text/plain')

  expect(exported).toBe(`- ${HOME_TOKEN}
  - a
    - a1
      - a2`)
```